### PR TITLE
Extract and refactor EditPostActivity.onEditorEvent

### LINKED ISSUE #10793: Extract and refactor OnEditorEvent
Extract and refactor EditPostActivity.onEditorEvent()

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #10793 

To test:
1. Open an editor
2. Type something
3. Select eg. Bold
4. Make sure the event gets tracked - check LogCat

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -3024,125 +3026,18 @@ public void onHtmlModeToggledInToolbar() {
...
-                currentStat = Stat.EDITOR_TAPPED_HEADING_5;
-                break;
-            case HEADING_6_BUTTON_TAPPED:
-                currentStat = Stat.EDITOR_TAPPED_HEADING_6;
-                break;
-            case HORIZONTAL_RULE_BUTTON_TAPPED:
-                currentStat = Stat.EDITOR_TAPPED_HORIZONTAL_RULE;
-                break;
-            case FORMAT_ALIGN_LEFT_BUTTON_TAPPED:
-                AnalyticsTracker.track(Stat.EDITOR_TAPPED_ALIGN_LEFT);
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: I've replaced cases where we use `AnalyticsTracker.track(XYZ)` with `currentStat = XYZ`. I believe the only thing that will change is that we'll include "editor=gutenberg/aztec" in its stats. Wdyt?

Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: yep, sounds about right (and correct) 👍

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java [hunk 2]
```diff
@@ -558,6 +559,10 @@ public void setTitle(CharSequence text) {
         mTitle.setText(text);
     }
 
+    @NonNull @Override public String getEditorName() {
```
Comment (planarvoid) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java]: NIT: move these to separate lines?

File: WordPress/src/main/java/org/wordpress/android/ui/posts/editor/EditorTracker.kt [hunk 3]
```diff
@@ -65,4 +67,42 @@ class EditorTracker @Inject constructor(private val context: Context) {
         }
         AnalyticsUtils.trackWithSiteDetails(stat, site, null)
     }
+
+    fun trackEditorEvent(event: TrackableEvent, editorName: String) {
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/editor/EditorTracker.kt]: Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10794/commits/1b42972457f3f71937e65a84b091487ad57111e0

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java (hunk 2 above)
Comment (malinajirka) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java]: Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10794/commits/5341b48d43033f78d44d7ea5baf62249f6f78898

**Comment by jkmassel** [comment]
Howdy folks! I'm freezing `13.7` today, so this is being bumped to `13.8`. If you'd like it to go out as part of `13.7`, please target `release/13.7` for this PR, and DM me once it's merged – I'll be happy to cut a new beta release!

**Comment by malinajirka** [comment]
Thanks for the review @planarvoid! It's ready for another round ;).

**Pull request merged**